lib/libzfs/libzfs_util.c

```diff
@@ -1203,11 +1203,12 @@ zcmd_write_nvlist_com(libzfs_handle_t *hdl, uint64_t *outnv, uint64_t *outlen,
     nvlist_t *nvl)
 {
 	char *packed;
+	size_t len;
 
-	size_t len = fnvlist_size(nvl);
+	verify(nvlist_size(nvl, &len, NV_ENCODE_XDR) == 0);
```

Comment:
You created an `fnvlist_size_xdr()` is there a reason not to use it here?

Reply:
IIRC (and this is probably lost in the various force pushes) I backed off using fnvlist_size_xdr() in some locations in order to try to isolate which change was causing the test timeouts.